DPB_converger.py

```diff
@@ -0,0 +1,363 @@
+    """swft command per file"""
+    file_command = 'swft.exe ' + file_path + ' --tau ' + '"' + tau + '" ' + '--out_values "all" '
+    file_command += '--exp_displacement ' + '"' + exp_displacement + '"' + ' '
+    file_command += '--exp_noise_rate ' + '"' + str(exp_noise_rate) + '"' + ' '
+    file_command += '--max_displacement ' + '"' + max_displacement + '"' + ' '
+    file_command += '--max_displacement_pp ' + '"' + max_displacement_pp + '"' + ' '
+    file_command += '--p_bleach ' + '"' + p_bleach + '"' + ' '
+    file_command += '--p_switch ' + '"' + p_switch + '"' + ' '
+    file_command += '--precision ' + '"' + str(precision) + '"' + ' '
+    file_command += '--diffraction_limit ' + '"' + diff_limit + '"' + ' '
+    return file_command
+
+
+def gather_Data(filepaths):
+    """
+    Collects the displacement and pBleach data of all notebook outputs in the given list of directories
+    """
+    #gathers a list of all relevant files
+    displacement_files = []
+    bleach_files = []
+    for working_directory in filepaths:
+        displacement_files += get_matching_files(working_directory, 'displacement', 'histogram')
+        bleach_files += get_matching_files(working_directory, 'p_bleach', 'histogram')
+    displacement_data = []
+    bleach_data = []
+    # reads the values and puts them into one list, then returns it
+    for file in displacement_files:
+        disp = float(open(file, 'r').readlines()[1].split('	')[0])
+        displacement_data.append(disp)
+    for file in bleach_files:
+        bleach = float(open(file, 'r').readlines()[1].split('	')[0])
+        bleach_data.append(bleach)
+    return displacement_data, bleach_data
+
+
+def write_swift(config_path, exp_displacement, p_bleach):
+    """
+    Effectively does what the main method of the write_swift_batch.py does,
+    however the parameters exp_displacement and p_bleach are no longer read from file but given as arguments
+    """
+    file_paths = []
+    n_file_paths = []
+    precision_vals = []
+    exp_noise_rate_vals = []
+    config = configparser.ConfigParser()
+    config.sections()
+    config.read(config_path)
+    """
+    The script expects the files precision and exp_noise_rate to be in the folder
+    \PreAnalysisParameter\parameters, unlike the write_swift_batch.py which allows for a directory to be specified.
+    However this is where BatchPreAnalysis places them.
+    """
+    try:
+        if len([key for key in config["INPUT_DIRS"]]):
+            for key in config["INPUT_DIRS"]:
+                file_paths_dir = get_loc_files(config["INPUT_DIRS"][key] + "\\cells\\tracks")
+                n_file_paths.append(len(file_paths_dir))
+                file_paths.extend(file_paths_dir)
+                precision_vals.append(
+                    config["INPUT_DIRS"][key] + '\\PreAnalysisParameters\\parameters\\precisions.txt')
+                assert (len(n_file_paths) == len(precision_vals))
```

Comment:
print out a meaningful error message for the assert call -> will be helpful for the user.
`assert (len(n_file_paths) == len(precision_vals), "message")`